src/components/result/FullListAndDetails.tsx

```diff
@@ -0,0 +1,80 @@
+import { FC, useCallback, useEffect } from "react";
+import { useSelector } from "react-redux";
+import { Box, Grid } from "@mui/material";
+import { useAppDispatch } from "../common/store/hooks";
+import store, {
+  getComponentState,
+  RootState,
+  searchQueryResult,
+} from "../common/store/store";
+import {
+  CollectionsQueryType,
+  createSearchParamFrom,
+  fetchResultAppendStore,
+  fetchResultWithStore,
+} from "../common/store/searchReducer";
+import { ParameterState } from "../common/store/componentParamReducer";
+import ListResultCard from "./ListResultCard";
+import { ResultCard } from "./ResultCards";
+import DetailSubtabBtn from "../common/buttons/DetailSubtabBtn";
+
+interface FullListAndDetailsProps extends ResultCard {}
+
+const FULL_LIST_PAGE_SIZE = 21;
+
+const FullListAndDetails: FC<FullListAndDetailsProps> = () => {
+  const dispatch = useAppDispatch();
+  const reduxContents = useSelector<RootState, CollectionsQueryType>(
+    searchQueryResult
+  );
+  const {
+    result: { collections },
+  } = reduxContents;
+
+  useEffect(() => {
+    if (collections.length < 20) {
+      const componentParam: ParameterState = getComponentState(
+        store.getState()
+      );
+      const paramPaged = createSearchParamFrom(componentParam, {
+        pagesize: FULL_LIST_PAGE_SIZE,
+      });
+      dispatch(fetchResultWithStore(paramPaged));
+    }
+    //only for fetching initial data
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const fetchMore = useCallback(async () => {
```

Comment:
I think the fetch is done on the top level so you do not need to fetch it here?

Reply:
thanks! I created a useFetchData hook for reusing in different components